extensions/ql-vscode/src/codeql-task-provider.ts

```diff
@@ -0,0 +1,172 @@
+import * as vscode from 'vscode';
+import { CodeQLCliServer } from './cli';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+import { getOnDiskWorkspaceFolders } from './helpers';
+import { DisposableObject } from './pure/disposable-object';
+import { UserCancellationException } from './commandRunner';
+
+interface CodeQLTaskDefinition extends vscode.TaskDefinition {
+  /**
+   * The command to run.
+   */
+  command: string;
+
+  /**
+   * Additional command arguments.
+   */
+  commandArgs: string[];
+
+  /**
+   * User-specified properties for the task.
+   */
+  prompts: Prompt[];
+}
+
+interface Prompt {
+  message: string;
+  values?: string[];
+}
+
+export class CodeQLTaskProvider implements vscode.TaskProvider {
+  static CodeQLType = 'codeql';
+  private tasks: vscode.Task[] | undefined;
+
+  constructor(private cliServer: CodeQLCliServer) { }
+
+  public provideTasks(): vscode.Task[] {
+    return this.getTasks();
+  }
+
+  public resolveTask(_task: vscode.Task): vscode.Task | undefined {
+    const command: string = _task.definition.command;
+    if (command) {
+      const definition: CodeQLTaskDefinition = <any>_task.definition;
+      return this.getTask(definition);
+    }
+    return undefined;
+  }
+
+  private getTasks(): vscode.Task[] {
+    if (this.tasks !== undefined) {
+      return this.tasks;
+    }
+    const cliCommands: CodeQLTaskDefinition[] = [
+      {
+        command: 'resolve qlpacks',
+        commandArgs: [
+          '--additional-packs',
+          getOnDiskWorkspaceFolders().join(path.delimiter),
+        ],
+        prompts: [],
+        type: CodeQLTaskProvider.CodeQLType,
+      },
+      {
+        command: 'pack install',
+        commandArgs: [],
+        prompts: [
+          {
+            message: 'Select the pack to install dependencies for',
+            values: getWorkspacePacks(),
+          },
+        ],
+        type: CodeQLTaskProvider.CodeQLType,
+      },
+      {
+        command: 'pack download',
+        commandArgs: [],
+        prompts: [
+          {
+            message: 'Enter the <package-scope/name[@version]> of the pack to download',
+          },
+        ],
```

Comment:
Rather than a prompt, we can hard code downloading all of the core language query packs and let the user manually override.